### PR TITLE
Add nginx forwarding

### PR DESCRIPTION
**Pull request by bendhouseart**
It's easier to track changes this way, not quite there yet but it's done for today (2024-10-02).

Will ping when I want a final review and merge.

### CODE REVIEW

File: example.env [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# set this to either nginx/production_nginx.conf  or nginx/development_nginx.conf
+NGINX_CONFIG_FILE=nginx/development_nginx.conf
+
+# Set the BRAINLIFE_PRODUCTION environment variable to true and use https"
+# (this will launch the server on port 443) and run with nginx/production_nginx.conf"
+# the following files are required for https to work:" 
+#   nginx/ssl/sslcert.cert"
+#   nginx/ssl/sslcert.key"
+#   nginx/ssl/sslpassword" # note you can disable the password requirement for nginx by omitting this field and commenting or deleting the requirement in the  nginx production NGINX_CONFIG_FILE
+# for https to work"
```
Comment (bendhouseart) [inline example.env]:
```suggestion
```

**Comment by bendhouseart** [comment]
Getting a bunch of these:

```brainlife_ezbids-handler  | 0|handler  | Error: Cannot find module 'mongoose'
brainlife_ezbids-handler  | 0|handler  | Require stack:
brainlife_ezbids-handler  | 0|handler  | - /app/api/models.js
brainlife_ezbids-handler  | 0|handler  | - /app/handler/handler.js
brainlife_ezbids-handler  | 0|handler  |     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1225:15)
brainlife_ezbids-handler  | 0|handler  |     at Module.Hook._require.Module.require (/usr/lib/node_modules/pm2/node_modules/require-in-the-middle/index.js:81:25)
brainlife_ezbids-handler  | 0|handler  |     at require (node:internal/modules/helpers:179:18)
brainlife_ezbids-handler  | 0|handler  |     at Object.<anonymous> (/app/api/models.ts:1:1)
brainlife_ezbids-handler  | 0|handler  |     at Module._compile (node:internal/modules/cjs/loader:1469:14)
brainlife_ezbids-handler  | 0|handler  |     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1548:10)
brainlife_ezbids-handler  | 0|handler  |     at Module.load (node:internal/modules/cjs/loader:1288:32)
brainlife_ezbids-handler  | 0|handler  |     at Function.Module._load (node:internal/modules/cjs/loader:1104:12)
brainlife_ezbids-handler  | 0|handler  |     at Module.require (node:internal/modules/cjs/loader:1311:19)
brainlife_ezbids-handler  | 0|handler  |     at Module.Hook._require.Module.require (/usr/lib/node_modules/pm2/node_modules/require-in-the-middle/index.js:101:39) {
brainlife_ezbids-handler  | 0|handler  |   code: 'MODULE_NOT_FOUND',
brainlife_ezbids-handler  | 0|handler  |   requireStack: [ '/app/api/models.js', '/app/handler/handler.js' ]
brainlife_ezbids-handler  | 0|handler  | }```

File: docker-compose.yml [hunk 2]
```diff
@@ -102,16 +102,15 @@ services:
...
             - 80:80
             - 443:443
         networks:
             - ezbids
         volumes: 
             - ./nginx/ssl:/etc/nginx/conf.d/ssl/
-            # replace production/development depending on whether you want ssl or not
-            - ./nginx/production_nginx.conf:/etc/nginx/conf.d/default.conf
+            # replace production/development depending on whether you want ssl or not, defaults to .nginx/development_nginx.conf
+            - ./${NGINX_CONFIG_FILE}:/etc/nginx/conf.d/default.conf
```
Comment (bendhouseart) [inline docker-compose.yml]:
```suggestion
            - ./${NGINX_CONFIG_FILE}:/etc/nginx/conf.d/nginx.conf
```

Comment (bendhouseart) [inline docker-compose.yml]:
```suggestion
            # replace production/development depending on whether you want ssl or not, defaults to .nginx/development_nginx.conf
            - ./${NGINX_CONFIG_FILE}:/etc/nginx/conf.d/nginx.conf
```

**Comment by bendhouseart** [comment]
@buti1021 this works pretty well, I haven't done much testing yet for http/non-production, but I'm happy with it otherwise. Take a look and let me know what you think. I'm sure there are bugs, but I think it's in a good enough state to push into ezbids proper.

File: ui/vite.config.ts
Comment (buti1021) [inline ui/vite.config.ts:1]: What is the intention behind those changes?

Comment (bendhouseart) [reply ui/vite.config.ts:1]: I think below was something that Anibal and I tried early on to get vite to stop refreshing. I'm not certain about what it's doing now, my thoughts were if I didn't look at it too hard it wouldn't wither and break.

**Comment by buti1021** [comment]
Hey @bendhouseart Anthony, I removed the nginx-development stuff, added some vars for the paths to the certs as I think those configuration should be configurable and used the existing compose profiles variable to make it possible to enable telemetry. Tested on our infrastructure :). Let me know if you are happy like this so we can look to get this merged upstream

**Comment by bendhouseart** [comment]
I'm most likely happy, but let me test it out to make sure it still works on my end before I commit to that. Will @ you once I've tested on my deployment.

**Comment by bendhouseart** [comment]
Tested by updating .env file with certs and/or to not use production. Works great for both!

image-qouqAayFQVCGJFyAtTTb

**Pull request merged**